src/cursor/cur_version.c

```diff
@@ -421,8 +421,8 @@ __curversion_next_single_key(WT_CURSOR *cursor)
                 }
 
                 if (F_ISSET(version_cursor, WT_CURVERSION_VISIBLE_ONLY) &&
-                  cbt->upd_value->tw.prepare) {
-                    if (!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw))
+                  WT_TIME_WINDOW_HAS_PREPARE(&(cbt->upd_value->tw))) {
+                    if (!WT_TIME_WINDOW_HAS_STOP_PREPARE(&cbt->upd_value->tw))
                         goto skip_on_page;
 
                     if (stop_txn == cbt->upd_value->tw.start_txn)
```

Comment:
This can be changed to check WT_TIME_WINDOW_HAS_START_PREPARE(&cbt->upd_value->tw)

Reply:
the original condition is `!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)` which checks for not have stop prepare though. just confirming if we should swap to `WT_TIME_WINDOW_HAS_START_PREPARE` or keep `!WT_TIME_WINDOW_HAS_STOP_PREPARE`

Reply:
nevermind, i misread the line... the full block should be:


```
if (!WT_TIME_WINDOW_HAS_STOP_PREPARE(&cbt->upd_value->tw))
                        goto skip_on_page;

 if (WT_TIME_WINDOW_HAS_START_PREPARE(&cbt->upd_value->tw))
...
```

Reply:
line 426 should be !WT_TIME_WINDOW_HAS_STOP_PREPARE and line 429 should be WT_TIME_WINDOW_HAS_START_PREPARE

Reply:
That's correct.

Reply:
I think this can be simplified to:
```suggestion
                    if (F_ISSET(version_cursor, WT_CURVERSION_VISIBLE_ONLY) {
                               if (!WT_TIME_WINDOW_HAS_STOP_PREPARE(&cbt->upd_value->tw))
                                      goto skip_on_page;
                               
                               if (WT_TIME_WINDOW_HAS_START_PREPARE(&cbt->upd_value->tw))
                                      goto skip_on_page;
                               
                               stop_txn = WT_TXN_MAX;
                               stop_ts = WT_TS_MAX;
                               durable_stop_ts = WT_TS_MAX;
                               version_prepare_state = 0;
                    } else
                               version_prepare_state = WT_TIME_WINDOW_HAS_PREPARE(&(cbt->upd_value->tw));
```

Reply:
I think we only want to assign `stop_txn` and `stop_ts` `durable_stop_ts` and to max if tw is prepared, so cant really remove the outer prepare check condition.

Actually I think the original version condition check is simple enough so would be better to keep them as is

Reply:
oh, yes, that's correct.

Reply:
I would say we still change this check

`if (stop_txn == cbt->upd_value->tw.start_txn)`. The intention of this check is not as clear as WT_TIME_WINDOW_HAS_START_PREPARE